### PR TITLE
Sor10874/real trace configs

### PR DESCRIPTION
**Pull request by sorenoid**
Related to issue: https://devtopia.esri.com/runtime/kotlin/issues/4583

### Description:

implement state for a selected trace configuration
implement data class representing a starting point from a Feature and UtilityElement.

### Pre-merge Checklist

- a [vTest](https://runtime-kotlin.esri.com/view/all/job/vtest/job/toolkit/) Job for this PR has been run
  - [x] link: https://runtime-kotlin.esri.com/view/vTest/job/vtest/job/toolkit/99/
- Unit and/or integration tests have been added to exercise this PR's logic, and the tests are passing:
  - [ ] Yes
  - [ ] No

### CODE REVIEW

File: toolkit/utilitynetworks/src/main/java/com/arcgismaps/toolkit/utilitynetworks/TraceState.kt [hunk 1]
```diff
@@ -82,45 +92,51 @@ public class TraceState(
...
 
-    private var utilityNetwork: UtilityNetwork? = null
 
     init {
         coroutineScope.launch {
-            arcGISMap.load().onSuccess {
-                arcGISMap.utilityNetworks.forEach {
-                    it.load()
-                }
+            arcGISMap.load()
```
Comment (puneet-pdx) [inline toolkit/utilitynetworks/src/main/java/com/arcgismaps/toolkit/utilitynetworks/TraceState.kt:115]: Shouldn't we be check for successful loading of the map here before loading the utilitynetworks?

File: toolkit/utilitynetworks/src/main/java/com/arcgismaps/toolkit/utilitynetworks/TraceState.kt [hunk 2]
```diff
@@ -159,6 +175,36 @@ public class TraceState(
...
+    internal fun removeStartingPoint(startingPoint: StartingPoint) {
+        _startingPoints.remove(startingPoint)
+    }
+
+    private suspend fun addStartingPoint(feature: ArcGISFeature) = try {
+        // TODO: add fraction-along to the element.
+        val utilityElement = utilityNetwork.createElementOrNull(feature)
+            ?: throw IllegalArgumentException("could not create utility element from ArcGISFeature")
+
+        // TODO: symbols are too small.
```
Comment (puneet-pdx) [inline toolkit/utilitynetworks/src/main/java/com/arcgismaps/toolkit/utilitynetworks/TraceState.kt:187]: How are we planning on addressing these Todos, should we create issues for them?

File: toolkit/utilitynetworks/src/main/java/com/arcgismaps/toolkit/utilitynetworks/ui/StartingPoint.kt [hunk 3]
```diff
@@ -101,7 +82,7 @@ private const val DELETE_THRESHOLD = 5f
  */
 @Composable
 internal fun StartingPoint(
```
Comment (puneet-pdx) [inline toolkit/utilitynetworks/src/main/java/com/arcgismaps/toolkit/utilitynetworks/ui/StartingPoint.kt:84]: Should we rename this to `StartingPointRow` so it is explicit what this composable represents ? 
We also have a StartingPoint data type defined. This rename will help differentiate between the two.

File: toolkit/utilitynetworks/src/main/java/com/arcgismaps/toolkit/utilitynetworks/ui/TraceOptionsScreen.kt [hunk 4]
```diff
@@ -132,17 +141,17 @@ internal fun TraceOptionsScreen(
...
  */
 @Composable
-private fun TraceConfiguration(utilityTraces: List<SelectableItem>) {
+private fun TraceConfigurations(configs: List<UtilityNamedTraceConfiguration>, selectedConfig: UtilityNamedTraceConfiguration?, onTraceSelected: (UtilityNamedTraceConfiguration)->Unit) {
     ExpandableCard(
         title = stringResource(id = R.string.trace_configuration),
         padding = 4.dp
     ) {
-        var selectedTrace: SelectableItem? by remember { mutableStateOf(null) }
+        //var selectedConfiguration: String by remember { mutableStateOf(selectedConfigName) }
```
Comment (puneet-pdx) [inline toolkit/utilitynetworks/src/main/java/com/arcgismaps/toolkit/utilitynetworks/ui/TraceOptionsScreen.kt:149]: unused code, could be removed?

File: toolkit/utilitynetworks/src/main/java/com/arcgismaps/toolkit/utilitynetworks/ui/TraceOptionsScreen.kt [hunk 5]
```diff
@@ -171,19 +180,16 @@ private fun TraceConfiguration(utilityTraces: List<SelectableItem>) {
  * @since 200.6.0
  */
 @Composable
-private fun StartingPointsEditor(showAddStartingPointScreen: () -> Unit) {
-    val startingPoints = remember { mutableStateListOf(StartingPointData(name = "Test Starting Point")) }
-    var counter by remember { mutableIntStateOf(1) }
-    ExpandableCard(
-        title = "${stringResource(id = R.string.starting_points)} (${counter})",
+private fun StartingPoints(startingPoints: List<StartingPoint>, showAddStartingPointScreen: () -> Unit, onStartingPointRemoved: (StartingPoint)->Unit) {
```
Comment (puneet-pdx) [inline toolkit/utilitynetworks/src/main/java/com/arcgismaps/toolkit/utilitynetworks/ui/TraceOptionsScreen.kt]:
```suggestion
private fun StartingPointsCard(startingPoints: List<StartingPoint>, showAddStartingPointScreen: () -> Unit, onStartingPointRemoved: (StartingPoint)->Unit) {
```

**Review by puneet-pdx — COMMENTED** [review]
@sorenoid looks good. Some minor comments

**Review by puneet-pdx — APPROVED** [review]
LGTM

**Comment by puneet-pdx** [comment]
@sorenoid I noticed when you tap on the map the features are displayed in the Starting point card, with their icons. But when you delete the features, the remaining features don't display the correct icon. Last item in the row takes up the icon of the item that was deleted. I will create an issue for it.

File: toolkit/utilitynetworks/src/main/java/com/arcgismaps/toolkit/utilitynetworks/TraceState.kt [hunk 6]
```diff
@@ -159,6 +176,34 @@ public class TraceState(
         }
     }
 
+    internal fun removeStartingPoint(startingPoint: StartingPoint) {
+        _startingPoints.remove(startingPoint)
+    }
+
+    private fun addStartingPoint(feature: ArcGISFeature) = try {
```
Comment (puneet-pdx) [inline toolkit/utilitynetworks/src/main/java/com/arcgismaps/toolkit/utilitynetworks/TraceState.kt:183]:
```suggestion
    private fun processAndAddStartingPoint(feature: ArcGISFeature) = try {
```

**Review by puneet-pdx — APPROVED** [review]
@sorenoid just another naming suggestion

**Review by eri9000 — COMMENTED** [review]
Good work 👍🏼 
a few suggestions

**Review by eri9000 — APPROVED** [review]
Good work!

**Pull request merged**